tests/testing_utils/files_and_dirs.py

```diff
@@ -31,6 +31,7 @@ def create_temp_file(suffix: str, dir_name: str, contents: List[str]) -> str:
 
 def create_named_file(file_name: str, dir_name: str, contents: List[str]):
     file_path = os.path.join(dir_name, file_name)
+    os.makedirs(os.path.dirname(file_path), exist_ok=True)
```

Comment:
why is this in this PR? it doesn't seem necessary for tests you're touching

Reply:
I split the main PR into 2 parts. This one involves all the files where snowCLI approval is required. This will be needed for the event sharing PR: please see https://github.com/snowflakedb/snowflake-cli/pull/1823/files for more info